src/folding/nova/circuits.rs

```diff
@@ -0,0 +1,275 @@
+{
+    fn new_variable<T: Borrow<CommittedInstance<C>>>(
+        cs: impl Into<Namespace<CF1<C>>>,
+        f: impl FnOnce() -> Result<T, SynthesisError>,
+        mode: AllocationMode,
+    ) -> Result<Self, SynthesisError> {
+        f().and_then(|val| {
+            let cs = cs.into();
+
+            let u = FpVar::<C::ScalarField>::new_variable(cs.clone(), || Ok(val.borrow().u), mode)?;
+            let x: Vec<FpVar<C::ScalarField>> =
+                Vec::new_variable(cs, || Ok(val.borrow().x.clone()), mode)?;
+
+            Ok(Self {
+                _c: PhantomData,
+                u,
+                x,
+            })
+        })
+    }
+}
+
+/// CommittedInstance on E2 contains the commitments to E and W, which are folded on the auxiliary
+/// curve constraints field (E2::Fr = E1::Fq).
+pub struct CommittedInstanceE2Var<C: CurveGroup, GC: CurveVar<C, CF2<C>>>
+where
+    for<'a> &'a GC: GroupOpsBounds<'a, C, GC>,
+{
+    _c: PhantomData<C>,
+    cmE: GC,
+    cmW: GC,
+}
+
+impl<C, GC> AllocVar<CommittedInstance<C>, CF2<C>> for CommittedInstanceE2Var<C, GC>
+where
+    C: CurveGroup,
+    GC: CurveVar<C, CF2<C>>,
+    for<'a> &'a GC: GroupOpsBounds<'a, C, GC>,
+{
+    fn new_variable<T: Borrow<CommittedInstance<C>>>(
+        cs: impl Into<Namespace<CF2<C>>>,
+        f: impl FnOnce() -> Result<T, SynthesisError>,
+        mode: AllocationMode,
+    ) -> Result<Self, SynthesisError> {
+        f().and_then(|val| {
+            let cs = cs.into();
+
+            let cmE = GC::new_variable(cs.clone(), || Ok(val.borrow().cmE), mode)?;
+            let cmW = GC::new_variable(cs.clone(), || Ok(val.borrow().cmW), mode)?;
+
+            Ok(Self {
+                _c: PhantomData,
+                cmE,
+                cmW,
+            })
+        })
+    }
+}
+
+/// Implements the circuit that does the checks of the Non-Interactive Folding Scheme Verifier
+/// described in section 4 of https://eprint.iacr.org/2021/370.pdf, where the cmE & cmW checks are
+/// delegated to the NIFSCycleFoldGadget.
+pub struct NIFSGadget<C: CurveGroup> {
+    _c: PhantomData<C>,
+}
+
+impl<C: CurveGroup> NIFSGadget<C>
+where
+    C: CurveGroup,
+{
+    /// Implements the constraints for NIFS.V for u and x, since cm(E) and cm(W) are delegated to
+    /// the CycleFold circuit.
+    pub fn verify(
+        r: FpVar<CF1<C>>,
+        ci1: CommittedInstanceE1Var<C>,
+        ci2: CommittedInstanceE1Var<C>,
+        ci3: CommittedInstanceE1Var<C>,
+    ) -> Result<(), SynthesisError> {
+        ci3.u.enforce_equal(&(ci1.u + r.clone() * ci2.u))?;
+
+        let x_rlc = ci1
+            .x
+            .iter()
+            .zip(ci2.x)
+            .map(|(a, b)| a + &r * &b)
+            .collect::<Vec<FpVar<CF1<C>>>>();
+        x_rlc.enforce_equal(&ci3.x)?;
+
+        Ok(())
+    }
+}
+
+/// NIFSCycleFoldGadget performs the Nova NIFS.V elliptic curve points relation checks in the other
+/// curve following CycleFold (https://eprint.iacr.org/2023/1192.pdf).
```

Comment:
What I'm seeing that arkworks does is using markdown syntax for the links: 
https://github.com/arkworks-rs/curves/tree/3fded1fbccf7c5491930dd8340c9b4567355d509/bn254/src/lib.rs#L15

In this specific line I was not intending to create a special link, just to have the link inside a parenthesis as it would be in a normal text. But, since we're at it, how do you see to use the markdown link syntax as the arkworks link provided?

Reply:
Ohh you can do markdown style too. But you weren't doing any of these. And rustdoc should spit a warning related to this.

Reply:
Yes, because here I was not trying to style the link, just leaving it inside a parenthesis as plaintext. But happy to update it using markdown. Will update it now.